presentation-compiler/test/dotty/tools/pc/tests/completion/CompletionSuite.scala

```diff
@@ -634,14 +634,40 @@ class CompletionSuite extends BaseCompletionSuite:
          |""".stripMargin
     )
 
+  @Test def patRecursive =
+    check(
+      s"""|object Main {
+          |  Option(List(Option(1))) match {
+          |    case Some(List(None, Som@@))
+          |}
+          |""".stripMargin,
+      """|Some(value) scala
+         |Some[A](value: A): Some[A]
```

Comment:
I'm not sure whether we should still show synthetic constructor / apply completions in this case. They are invalid here. 

Completions for completions is a different story though, as we have no logic to detect whether it has nested members yet, so this is another improvement that could be made in the future.

If you want to remove `Some[A](value: A): Some[A]` I suppose it can be achieved by adding unapply cases here:
```scala
//file: Completions.scala
  private lazy val shouldAddSnippet =
    path match
      /* In case of `method@@()` we should not add snippets and the path
       * will contain apply as the parent of the current tree.
       */
      case (fun) :: (appl: GenericApply) :: _ if appl.fun == fun =>
        false
      case _ :: (withcursor @ Select(fun, name)) :: (appl: GenericApply) :: _
          if appl.fun == withcursor && name.decoded == Cursor.value =>
        false
      case (_: (Import | Export)) :: _ => false
      case _ :: (_: (Import | Export)) :: _ => false
      case _ => true
```